sklearn/tree/tests/test_tree.py

```diff
@@ -2135,6 +2139,10 @@ def test_decision_tree_regressor_sample_weight_consistentcy(criterion):
     # make it positive in order to work also for poisson criterion
     y += np.min(y) + 0.1
 
+    # convert dtype
+    X = X.astype(global_dtype)
+    y = y.astype(global_dtype)
```

Comment:
This can prevent an extra copy. I just changed the description of the issue accordingly to have it indicated.
```suggestion
    X = X.astype(global_dtype, copy=False)
    y = y.astype(global_dtype, copy=False)
```